hubploy/auth.py

```diff
@@ -216,32 +249,66 @@ def cluster_auth_gcloud(deployment, project, cluster, zone, service_key):
     yield
 
 
-def cluster_auth_aws(deployment, project, cluster, zone, service_key):
+def cluster_auth_aws(deployment, project, cluster, zone, service_key=None, role=None):
     """
-    Setup AWS authentication with service_key
+    Setup AWS authentication with service_key or with a role
 
     This changes *global machine state* on what current kubernetes cluster is!
     """
-    # Get credentials from standard location
-    service_key_path = os.path.join(
-        'deployments', deployment, 'secrets', service_key
-    )
 
-    original_credential_file_loc = os.environ.get("AWS_SHARED_CREDENTIALS_FILE", None)
+    if not service_key and not role:
+        raise Exception('AWS authentication requires either a service key or the use of a role')
+
+    def update_kubeconfig():
+        subprocess.check_call([
+        'aws', 'eks', 'update-kubeconfig',
+        '--name', cluster, '--region', zone
+    ])
 
     try:
+        if service_key:
+            # Get credentials from standard location
+            service_key_path = os.path.join(
+                'deployments', deployment, 'secrets', service_key
+            )
+
+            original_credential_file_loc = os.environ.get("AWS_SHARED_CREDENTIALS_FILE", None)
 
-        # Set env variable for credential file location
-        os.environ["AWS_SHARED_CREDENTIALS_FILE"] = service_key_path
+            # Set env variable for credential file location
+            os.environ["AWS_SHARED_CREDENTIALS_FILE"] = service_key_path
 
-        subprocess.check_call(['aws', 'eks', 'update-kubeconfig',
-                               '--name', cluster, '--region', zone])
+            update_kubeconfig()
+
+        if role:
+            # this doesn't come back in the sts client response
+            role_session_name = 'cluster'
+
+            sts_client = boto3.client('sts')
+            assumed_role_object = sts_client.assume_role(
+                RoleArn = role,
```

Comment:
[pep-8](https://www.python.org/dev/peps/pep-0008/) suggests no spaces around `=` signs for function arguments. Can you remove the extra spaces?